src/mesh/mesh.cxx

```diff
@@ -433,11 +438,13 @@ std::shared_ptr<Coordinates> Mesh::createDefaultCoordinates(const CELL_LOC locat
 
   if (location == CELL_CENTRE || location == CELL_DEFAULT) {
     // Initialize coordinates from input
-    return std::make_shared<Coordinates>(this, options);
+    auto result = std::make_shared<Coordinates>(this, options);
+    return result;
   } else {
     // Interpolate coordinates from CELL_CENTRE version
-    return std::make_shared<Coordinates>(this, options, location,
+    auto result = std::make_shared<Coordinates>(this, options, location,
         getCoordinates(CELL_CENTRE), force_interpolate_from_centre);
+    return result;
```

Comment:
Was this change because you were going to do `result.geometry()` before returning it? Was there an issue with that?

Reply:
Yes and yes. 6a9e00b1553ed9414402677937aef53c5e4fe45b moved the `geometry()` call to `getCoordinatesSmart()` to make sure that `field.getCoordinates()` never returns `nullptr`.